client/src/components/DropdownMenu.tsx

```diff
@@ -1,22 +1,44 @@
 import classNames from "classnames";
 import _ from "lodash";
-import React from "react";
+import React, { BaseSyntheticEvent } from "react";
 
-import "./DropdownMenu.scss";
 import { trivial_text_maker } from "src/models/text.js";
 
-export class DropdownMenu extends React.Component {
-  constructor(props) {
+import "./DropdownMenu.scss";
+
+interface DropdownMenuProps {
+  dropdown_content: React.ReactNode;
+  opened_button_class_name: string;
+  closed_button_class_name: string;
+  dropdown_trigger_txt: React.ReactNode | string;
+  dropdown_content_class_name?: string;
+  button_description?: string;
+  dropdown_a11y_txt?: string;
+}
+
+type DropdownMenuState = {
+  is_open: boolean;
+};
+
+export class DropdownMenu extends React.Component<
+  DropdownMenuProps,
+  DropdownMenuState
+> {
+  dropdown_ref = React.createRef<HTMLDivElement>();
+  toggle_dropdown_button = React.createRef<HTMLDivElement>();
+  dropdown_area = React.createRef<HTMLDivElement>();
+
+  constructor(props: DropdownMenuProps) {
     super(props);
     this.state = {
       is_open: false,
     };
```

Comment:
Another bit of annoyance: can't use that experimental syntax to initalize state outside of constructor so we have worst of both worlds😔